### PR TITLE
e2e/framework: remove direct imports to /pkg/api/v1/pod

### PR DESCRIPTION
**Pull request by gavinfish**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Removes imports to pkg/api/v1/pod in e2e framework as part of moving it to staging.

**Which issue(s) this PR fixes**:

Xref #74352

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
none
```

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -54,14 +54,15 @@ import (
...
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 	clientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/kubernetes/scheme"
 	"k8s.io/client-go/rest"
 	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 	watchtools "k8s.io/client-go/tools/watch"
-	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"k8s.io/component-base/featuregate"
```
Comment (tanjunchen) [inline test/e2e/framework/util.go:65]: /cc  @oomichi 

could you review this pr due to long time . Thanks

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -738,6 +739,66 @@ func (f *Framework) testContainerOutputMatcher(scenarioName string,
...
+
+// ContainerVisitor is called with each container spec, and returns true
+// if visiting should continue.
+type ContainerVisitor func(container *v1.Container, containerType ContainerType) (shouldContinue bool)
+
+// visitContainers invokes the visitor function with a pointer to every container
+// spec in the given pod spec with type set in mask. If visitor returns false,
+// visiting is short-circuited. visitContainers returns true if visiting completes,
+// false if visiting was short-circuited.
+func visitContainers(podSpec *v1.PodSpec, mask ContainerType, visitor ContainerVisitor) bool {
```
Comment (oomichi) [inline test/e2e/framework/util.go:777]: ditto: `NOTE`

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -738,6 +739,66 @@ func (f *Framework) testContainerOutputMatcher(scenarioName string,
...
+	// Containers is for normal containers
+	Containers ContainerType = 1 << iota
+	// InitContainers is for init containers
+	InitContainers
+	// EphemeralContainers is for ephemeral containers
+	EphemeralContainers
+)
+
+// AllContainers specifies that all containers be visited
+const AllContainers ContainerType = InitContainers | Containers | EphemeralContainers
```
[the remark this answers is not in the document]
Comment (gavinfish) [reply test/e2e/framework/util.go]: done

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -738,6 +739,66 @@ func (f *Framework) testContainerOutputMatcher(scenarioName string,
...
+
+// allFeatureEnabledContainers returns a ContainerType mask which includes all container
+// types except for the ones guarded by feature gate.
+func allFeatureEnabledContainers() ContainerType {
```
[the remark this answers is not in the document]
Comment (gavinfish) [reply test/e2e/framework/util.go:763]: done

**Review by oomichi — COMMENTED** [review]
Thanks for updating

/lgtm

File: test/e2e/framework/util.go (hunk 2 above)
Comment (YuikoTakada) [reply test/e2e/framework/util.go:777]: func VisitContainers in  pkg/api/v1/pod will be removed later or remain?

**Pull request merged**